src/middlewared/middlewared/plugins/etc.py

```diff
@@ -266,6 +272,9 @@ def __init__(self, *args, **kwargs):
             if rendered is None:
                 continue
 
+            if not os.path.exists(outfile.rsplit('/', 1)[0]):
```

Comment:
I just now saw this, I think we can use os.path.dirname instead, which is an "official" way to do what you did